dts/bindings/rtc/maxim,ds3231-rtc.yaml

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2024 Gergo Vari <work@gergovari.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Maxim DS3231 I2C RTC/TCXO
```

Comment:
Could you expand the descriptions to provide example devicetree layouts for the MFD device? Similar to https://github.com/zephyrproject-rtos/zephyr/blob/feaf21f80cb6ddb5dafa53056154707c18c6b2c2/dts/bindings/comparator/nordic%2Cnrf-comp.yaml#L10-L17 

I believe this may be essential for devs to understand how to use the device :)